geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-api/src/main/scala/org/locationtech/geomesa/fs/storage/api/StorageMetadata.scala

```diff
@@ -100,6 +100,19 @@ trait StorageMetadata extends Compactable with Closeable {
     */
   def removePartition(partition: PartitionMetadata): Unit
 
+  /**
+   * Overwrite any existing partitions
+   *
+   * @param partitions partitions
+   */
+  def setPartitions(partitions: Seq[PartitionMetadata]): Unit =
+    throw new NotImplementedError() // TODO remove default impl in next major release
```

Comment:
Is there already a deprecation on the interface?  Also, do we already have a ticket for that particular deprecation?

Reply:
this is a new method, not a deprecation. i added a default impl so that any existing classes would still compile against it.

Reply:
Ah, so the TODO is to add implementations to the other classes which inherit from this interface?

Reply:
all of our classes already have impls, it's just in case external people have added impls. with the default NotimplementedError, it doesn't break semver, but once we remove that it will (at least that's my interpretation...) so the TODO is for the api breaking change that we can't make until a major release

Reply:
Ah!  Got it!

Some day, I want to see a minimal GitHub/Java/Scala project to show all these little version / API upgrade situations.